jme3-lwjgl3/src/main/java/com/jme3/system/lwjgl/LwjglWindow.java

```diff
@@ -1,646 +1,641 @@
-            final BufferedImage image = (BufferedImage) images[i];
-            out[i] = imageToGLFWImage(image);
-        }
-
-        return out;
-    }
-
-    /**
-     * Convert the {@link BufferedImage} to the {@link GLFWImage}.
-     */
-    private GLFWImage imageToGLFWImage(BufferedImage image) {
-
-        if (image.getType() != BufferedImage.TYPE_INT_ARGB_PRE) {
-
-            final BufferedImage convertedImage = new BufferedImage(image.getWidth(), image.getHeight(), BufferedImage.TYPE_INT_ARGB_PRE);
-            final Graphics2D graphics = convertedImage.createGraphics();
-
-            final int targetWidth = image.getWidth();
-            final int targetHeight = image.getHeight();
-
-            graphics.drawImage(image, 0, 0, targetWidth, targetHeight, null);
-            graphics.dispose();
-
-            image = convertedImage;
-        }
-
-        final ByteBuffer buffer = BufferUtils.createByteBuffer(image.getWidth() * image.getHeight() * 4);
-
-        for (int i = 0; i < image.getHeight(); i++) {
-            for (int j = 0; j < image.getWidth(); j++) {
-                int colorSpace = image.getRGB(j, i);
-                buffer.put((byte) ((colorSpace << 8) >> 24));
-                buffer.put((byte) ((colorSpace << 16) >> 24));
-                buffer.put((byte) ((colorSpace << 24) >> 24));
-                buffer.put((byte) (colorSpace >> 24));
-            }
-        }
-
-        buffer.flip();
-
-        final GLFWImage result = GLFWImage.create();
-        result.set(image.getWidth(), image.getHeight(), buffer);
-
-        return result;
-    }
-
-    /**
-     * Destroy the context.
-     */
-    protected void destroyContext() {
-        try {
-            if (renderer != null) {
-                renderer.cleanup();
-            }
-
-            if (errorCallback != null) {
-
-                // We need to specifically set this to null as we might set a new callback before we reinit GLFW
-                glfwSetErrorCallback(null);
-
-                errorCallback.close();
-                errorCallback = null;
-            }
-
-            if (windowSizeCallback != null) {
-                windowSizeCallback.close();
-                windowSizeCallback = null;
-            }
-
-            if (windowFocusCallback != null) {
-                windowFocusCallback.close();
-                windowFocusCallback = null;
-            }
-
-            if (window != NULL) {
-                glfwDestroyWindow(window);
-                window = NULL;
-            }
-
-            glfwTerminate();
-        } catch (final Exception ex) {
-            listener.handleError("Failed to destroy context", ex);
-        }
-    }
-
-    @Override
-    public void create(boolean waitFor) {
-        if (created.get()) {
-            LOGGER.warning("create() called when display is already created!");
-            return;
-        }
-
-        // NOTE: this is required for Mac OS X!
-        mainThread = Thread.currentThread();
```

Comment:
I know what they are doing here and your method might break OSX support. Do some research on the hub please:
For OpenGL to work on Mac OS X it has to be started on the MainThread (which is the Thread the Application is started with, hence Thread.currentThread(). Your change destroys this. So it changes semantics.

WaitFor() might be useless because if you're already on the MainThread, you don't have to wait for it.
But it's an inconsistent behavior as lwjgl2 probably returns from the Application#start() call.
So a proper solution would be to document this or provide AppSettings to enable "forking", unless on OS X.
Or always use the CurrentThread, which to me is expectable if you hand control over to a framework.

Reply:
Would it be better to check if we are on OS X, then ignore the wiatFor. Also, does this mean that `jme-lwjgl` does not work on OS X?

Reply:
Yes, this is probably better, but I don't like it cause we'd have an OS dependant behavior then. I need to triple check that with lwjgl3, but as pointed out in #1193, VR has the same problem.

lwjgl2 however is capable of working on Mac OS X, which is related to the way it creates the window, I guess. Which is why we need to double check things and re-check the above assumption. I guess I need to try that on mac.